File: BarCode/app/src/main/java/com/example/barcode/MainActivity.java
package com.example.barcode;

import android.content.Context;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.text.Editable;
import android.text.TextWatcher;
import android.util.Log;
import android.view.KeyEvent;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    RecyclerView recyclerView;
    RecyclerView.Adapter adapter;
    RecyclerView.LayoutManager layoutManager;
    Button button;
    ArrayList<Model> models;


    String Barcode = "";


    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        button = findViewById(R.id.button);
        models = new ArrayList<>();

        recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setHasFixedSize(true);
        layoutManager = new LinearLayoutManager(this);
        adapter = new MyAdapter(models, this);
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setAdapter(adapter);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                if (!Barcode.isEmpty()) {
                    valid(Barcode);
                    Toast.makeText(MainActivity.this,"Product Scanned",Toast.LENGTH_SHORT).show();
                    Barcode = "";
                }
            }
        });
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {

        char pressedKey = (char) event.getUnicodeChar();
        Barcode += "" + pressedKey;
        return true;
    }

    void valid(String name) {

        models.add(new Model(name));
        adapter.notifyDataSetChanged();
    }

}

File: RecyclerView/dynamic section recyclerview/app/src/main/java/com/alvin/dynamicsecctionrecyclerview/MainActivity.kt
package com.alvin.dynamicsecctionrecyclerview

import android.os.Bundle
import android.text.TextUtils
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import java.util.*
import kotlin.collections.ArrayList
import androidx.recyclerview.widget.ItemTouchHelper


class MainActivity : AppCompatActivity() {

    lateinit var rv: RecyclerView
    lateinit var mAdapter: Adapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        rv = findViewById(R.id.rv)

        val list = ArrayList<Model>()


        list.add(Model("One", true))
        list.add(Model("A", false))
        list.add(Model("A", false))
        list.add(Model("A", false))
        list.add(Model("A", false))
        list.add(Model("A", false))

        list.add(Model("Two", true))
        list.add(Model("B", false))
        list.add(Model("B", false))
        list.add(Model("B", false))
        list.add(Model("B", false))
        list.add(Model("B", false))

        list.add(Model("Three", true))
        list.add(Model("C", false))
        list.add(Model("C", false))
        list.add(Model("C", false))
        list.add(Model("C", false))
        list.add(Model("C", false))

        list.add(Model("Four", true))
        list.add(Model("d", false))
        list.add(Model("d", false))
        list.add(Model("d", false))
        list.add(Model("d", false))
        list.add(Model("d", false))

        rv.setHasFixedSize(true)
        val mLayoutManager = LinearLayoutManager(this)
        rv.layoutManager = mLayoutManager
        mAdapter = Adapter(this, list)
        rv.adapter = mAdapter

    }
}
